### ISSUE TITLE
validation set and warm start

### ISSUE THREAD

**Issue by az-ja** [princeton-vl/RAFT#62]
Hello,

I have 2 (other ;) ) questions regarding your great work.

1. This method has many degrees of freedom. Choosing different hyperparameters, the datasets to train with in each stages of training, and many other things. I was curious, what validation set did you use so that you get great (Sintel) test results?
Usually the training error is not a good indicator in these kinds of approaches. For example, the choice of using a warm start or not. If you use the warm start on the training set, you get worse results on the training set. So what would encourage you to use the warm start for the test set? 

2. To train the network, this approach uses many training samples. Each of these samples consist of 2 images and the gt flow. Therefore the network is not trained in a "warm" manner. Then while testing, we enable the "warm-start". So the point of warm start is just to set a better initialization. Is that correct? I would be interested to know, how the network generalizes nicely in this context, because initially, the network learned to estimate the flow from scratch and not just refining what is given. Do you think this generalization is caused by the iterative scheme of the model? I find it a bit confusing. If we evaluate the model raft_things.pth on sintel (training), the warm results are not better (slightly better on clean and slightly worse on final) than the "cold" ones.

Thank you very much for your precious time and stay healthy.
Azin

**Comment by zachteed** [comment]
Hi Azin,

1. I used several different validation splits on Sintel to avoid over-tuning to a single split. One split I used was validation={ambush_2, ambush_6, bamboo_2, cave_4, market_6, temple_2} which was proposed by MaskFlowNet. When submitting to the test set, I used all of the Sintel train sequences.

2. The warm-start is possible due to the iterative nature of the network. I did notice a small improvement from using warm-start in our validation setup (about .05 pixels), which was what motivated me to submit it to the test set. The difference on the test set however, is much greater (about .3 pixels) which was surprising and I'm not really sure of the reason.

I did find that performance could be slightly improved by training with "warm-start", where flow was randomly initialized from previous flow estimates with some probability. However, this significantly complicates training. I ended up not using this in favor of simpler training
